tests/nimatro.py

```diff
@@ -0,0 +1,45 @@
+import unittest
+
+from utils.channel_access import ChannelAccess
+from utils.ioc_launcher import get_default_ioc_dir
+from utils.test_modes import TestModes
+
+
+DEVICE_PREFIX = "NIMATRO_01"
+
+IOCS = [
+    {
+        "name": DEVICE_PREFIX,
+        "directory": get_default_ioc_dir("NIMATRO"),
+        "macros": {},
+    },
+]
+
+TEST_MODES = [TestModes.RECSIM, TestModes.DEVSIM]
+
+
+class NimatroTests(unittest.TestCase):
+    """
+    Tests for the NIMA Trough IOC.
+    """
+    def setUp(self):
+        self.ca = ChannelAccess(device_prefix=DEVICE_PREFIX)
+        self.ca.assert_that_pv_exists("CONTROL:START", timeout=30)
+
+    def test_GIVEN_running_ioc_WHEN_set_target_area_sp_THEN_target_area_updated(self):
+        expected_value = 41.15
```

Comment:
What about invalid values. There were some limits set in the DB file for various things. We should have both positive and negative test cases.

Reply:
Updated tests so that a range of values are tested, as well as out of bounds tests. Updated the `assert integer within range` channel access function to be more general and simply check the bounds for upper and lower limits.